tensilelite/Tensile/SolutionStructs.py

```diff
@@ -1917,9 +1918,12 @@ def isDirectToVgprDoable(state, tc):
       reject(state, "DirectToVgpr%c does not supports InnerUnroll>1"%(tc))
       return False
 
-    # Reject TLU = UnrollMajorLDS (B only)
-    if tc == 'B' and (state["ProblemType"]["TLUA"] == state["UnrollMajorLDSA"] or state["ProblemType"]["TLUB"] == state["UnrollMajorLDSB"]):
-      reject(state, "DirectToVgpr%c does not supports TLU = UnrollMajorLDS"%(tc))
+    # Reject TLU = UnrollMajorLDS
+    if tc == 'A' and (state["ProblemType"]["TLUA"] == state["UnrollMajorLDSA"]):
+      reject(state, "DirectToVgprA does not supports TLUA = UnrollMajorLDSA")
+      return False
+    if tc == 'B' and (state["ProblemType"]["TLUB"] == state["UnrollMajorLDSB"]):
+      reject(state, "DirectToVgprB does not supports TLUB = UnrollMajorLDSB")
       return False
```

Comment:
Separated the cases of A and B, Tests passed.

Reply:
Follow this way to reduce duplicated code: state["ProblemType"]["TLU%c"%tc])

Reply:
done